cafeteria/src/main/java/org/itt/dao/DataBaseConnector.java

```diff
@@ -0,0 +1,54 @@
+package org.itt.dao;
+
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class DataBaseConnector {
+    private static volatile DataBaseConnector instance = null;
+    private static Connection connection;
+
+    private DataBaseConnector() {
+    }
+
+    public static DataBaseConnector getInstance() throws SQLException, ClassNotFoundException {
+        if (instance == null) {
+            synchronized (DataBaseConnector.class) {
+                if (instance == null) {
+                    try {
+                        Properties props = new Properties();
+                        try (InputStream input = DataBaseConnector.class.getClassLoader().getResourceAsStream("db.properties")) {
+                            props.load(input);
+                        }
+                        String url = props.getProperty("db.url");
+                        String username = props.getProperty("db.username");
+                        String password = props.getProperty("db.password");
+
+                        Class.forName("com.mysql.cj.jdbc.Driver");
+                        connection = DriverManager.getConnection(url, username, password);
+                        instance = new DataBaseConnector();
+                    } catch (Exception e) {
+                        throw new SQLException("Unable to connect to the database", e);
```

Comment:
Would any exception caught here always be of type SQLException? If not, we should try not to alter the exceptions being thrown.